datafusion/sql/src/expr/mod.rs

```diff
@@ -1017,6 +979,71 @@ impl<'a, S: ContextProvider> SqlToRel<'a, S> {
     }
 }
 
+pub struct ArrayFunctionPlanner {
```

Comment:
My thinking is that this `ArrayFunctionPlanner` would be in the array_functions crate and thus avoid the implicit dependency on names of functions